### PR TITLE
Fix asan test failures

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -1726,6 +1726,8 @@ def include_and_export(name):
 
     if 'address' in sanitize:
       shared.Settings.USE_ASAN = 1
+      if not shared.Settings.UBSAN_RUNTIME:
+        shared.Settings.UBSAN_RUNTIME = 2
```
Comment (kripken) [inline emcc.py:1730]: maybe a comment? i'm not sure why this makes sense to do.

File: src/preamble.js [hunk 2]
```diff
@@ -752,15 +752,17 @@ function instrumentWasmExportsWithAbort(exports) {
   for (var name in exports) {
     (function(name) {
       var original = exports[name];
-      
+
       // Wrap all functions, copy the other symbols.
-      if (typeof original === 'function') {
+      // Don't wrap the asan helper functions since they are used during
+      // checkStackCookie and would causes our tests to fail.
```
Comment (kripken) [inline src/preamble.js:758]:
```suggestion
      // checkStackCookie which would causes our tests to fail.
```

Comment (kripken) [reply src/preamble.js:758]: Why would they fail? Does asan call them after exit?

Comment (sbc100) [reply src/preamble.js:758]: The problem is test_abort_on_exceptions:

```
      addOnPostRun(function() {                                                                         
        try {                                                                                           
          // Crash the program                                                                          
          _crash();                                                                                     
        }                                                                                               
        catch(e) {                                                                                      
          // Catch the abort                                                                            
          out(true);                                                                                    
        }                                                                                               
                                                                                                        
        out("again");                                                                                   
                                                                                                        
        try {                                                                                           
          // Try executing some function again                                                          
          _crash();                                                                                     
        }                                                                                               
        catch(e) {                                                                                      
          // Make sure it failed with the expected exception                                            
          out(e === "program has already aborted!");                                                    
        }                                                                                               
      });    
```

It tries to trigger the exception which makes the module unusable, but handle the exception and exit cleanly.

The problem is that checkStackCookie always runs at the end of `run()` which is normally harmless, but in asan mode it calls back into the module.

The alternative I guess would be to have checkStackCookie check if the module has crashed in just NOP if it has?

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:1730]: I actually don't know why its needed, I'm just preserving the behaviour that was previously in system_libs.py where USE_ASAN implies ubsan.   The only difference is that by doing it this way I prevent two different versions of ubsan from being included at the same time.

File: src/preamble.js (hunk 2 above)
Comment (sbc100) [reply src/preamble.js]: Ok that worked!  Maybe better this way?

Comment (kripken) [reply src/preamble.js]: Yes, I think that's more correct.

**Review by kripken — APPROVED** [review]

**Pull request merged**